documentation/specs/proposed/evaluation-perf.md

```diff
@@ -0,0 +1,24 @@
+# Evaluation performance investigations
+In the current effort to improve performance of MSBuild, we ideantified the evaluation as one of the focus areas of this effort. Evalution is the ifrst step when building, and it determines references, how projects are connected and what needs to be build. Because of this it runs in every single build, be it Design-time builds in Visual Studio, up-to-date builds or full builds.
+
+## Description
+Current performance state of evaluation is mostly unkown, as it is not measured in any ways by the team. As such, we are unsure which specific areas can be improve. The investigation about this is necessary so we can identify weaknesses, and possible fixes.
+
+ - We could do profiling
+ - Jit compilation of MSBuild itself. 
+ - We could cache at eval
+
+ Constraint - needs to work as it does today, but fast. We can break some edge cases.
+
+## Goals and Motivation
+We are trying to make evaluation phase of the build more performant, since it is almost always executed any performance gain becomes noticeable. A performant evaluation phase would decrease build times in general, in CI cases it frees up resources, and in individual cases it can increase dev-loop performance by making up-to-date and incremental builds go faster.
+
+In this moment we are still in investigation phase, the obejective is to add code markers, so we can idetentify low hanging fixes, and improvement areas when testing builds within PerfStar.
```

Comment:
The markers are already there for the most part, is there something specific we're planning to add?